### PR TITLE
[SC-89] August 30 spell

### PR DESCRIPTION
**Pull request by barrutko**
Simulation of the mainnet spell execution on a Tenderly fork:
https://dashboard.tenderly.co/phoenixlabs/sparklend/fork/15a0293a-549e-4ef1-9167-f0dddad7eef4/simulation/8ae7d15e-1d5c-47f1-b80c-1dbd08054d81/

### CODE REVIEW

File: src/proposals/20230830/SparkEthereum_20230830.sol [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    {
+        IEngine.RateStrategyUpdate[] memory ratesUpdate = new IEngine.RateStrategyUpdate[](1);
+
+        Rates.RateStrategyParams memory weth = LISTING_ENGINE
+            .RATE_STRATEGIES_FACTORY()
+            .getStrategyDataOfAsset(WETH);
+
+        weth.optimalUsageRatio = NEW_WETH_OPTIMAL_USAGE_RATIO;
+        weth.variableRateSlope1 = NEW_WETH_VARIABLE_RATE_SLOPE_1;
+        weth.variableRateSlope2 = NEW_WETH_VARIABLE_RATE_SLOPE_2;
```
Comment (lucas-manuel) [inline src/proposals/20230830/SparkEthereum_20230830.sol:36]:
```suggestion
        weth.optimalUsageRatio  = NEW_WETH_OPTIMAL_USAGE_RATIO;
        weth.variableRateSlope1 = NEW_WETH_VARIABLE_RATE_SLOPE_1;
        weth.variableRateSlope2 = NEW_WETH_VARIABLE_RATE_SLOPE_2;
```

File: src/proposals/20230830/SparkEthereum_20230830.sol [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ */
+contract SparkEthereum_20230830 is SparkPayloadEthereum {
+
+    address public constant WETH                           = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    uint256 public constant NEW_WETH_OPTIMAL_USAGE_RATIO   = 0.90e27;
+    uint256 public constant NEW_WETH_VARIABLE_RATE_SLOPE_1 = 0.028e27;
+    uint256 public constant NEW_WETH_VARIABLE_RATE_SLOPE_2 = 1.20e27;
+
+    address public constant wstETH                         = 0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0;
+    uint256 public constant NEW_WSTETH_SUPPLY_CAP          = 400_000;
```
Comment (lucas-manuel) [inline src/proposals/20230830/SparkEthereum_20230830.sol:23]:
```suggestion
    address public constant wstETH                = 0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0;
    uint256 public constant NEW_WSTETH_SUPPLY_CAP = 400_000;
```

File: src/proposals/20230830/SparkEthereum_20230830.sol [hunk 3]
```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.10;
+
+import { SparkPayloadEthereum, IEngine, Rates, EngineFlags } from '../../SparkPayloadEthereum.sol';
+
+import { IPoolAddressesProvider } from 'aave-v3-core/contracts/interfaces/IPoolAddressesProvider.sol';
+
+/**
+ * @title  August 30, 2023 Spark Ethereum Proposal - Update ETH Kink, Update ETHn Kink Rate, Update wstETH Supply Cap
```
Comment (lucas-manuel) [inline src/proposals/20230830/SparkEthereum_20230830.sol:9]: `Update ETH utilization rate parameters, increase wstETH supply cap`

File: src/proposals/20230830/SparkEthereum_20230830.sol [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+ * @author Phoenix Labs
+ * @dev    This proposal updates ETH market parameters and raises wstETH supply cap
```
Comment (lucas-manuel) [inline src/proposals/20230830/SparkEthereum_20230830.sol:11]: Describe specific param changes here: variableRateSlope1, variableRateSlope2, optimalUsageRation, and supplyCap for wstETH

File: src/proposals/20230830/SparkEthereum_20230830.sol [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+ * Vote:   N/A
+ */
+contract SparkEthereum_20230830 is SparkPayloadEthereum {
+
+    address public constant WETH                           = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    uint256 public constant NEW_WETH_OPTIMAL_USAGE_RATIO   = 0.90e27;
+    uint256 public constant NEW_WETH_VARIABLE_RATE_SLOPE_1 = 0.028e27;
+    uint256 public constant NEW_WETH_VARIABLE_RATE_SLOPE_2 = 1.20e27;
+
+    address public constant wstETH                         = 0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0;
```
Comment (lucas-manuel) [inline src/proposals/20230830/SparkEthereum_20230830.sol:22]: WSTETH

File: src/proposals/20230830/SparkEthereum_20230830.sol (hunk 2 above)
Comment (lucas-manuel) [reply src/proposals/20230830/SparkEthereum_20230830.sol:23]: Group addresses and uints

File: src/proposals/20230830/SparkEthereum_20230830.sol [hunk 6]
```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.10;
+
+import { SparkPayloadEthereum, IEngine, Rates, EngineFlags } from '../../SparkPayloadEthereum.sol';
+
+import { IPoolAddressesProvider } from 'aave-v3-core/contracts/interfaces/IPoolAddressesProvider.sol';
```
Comment (lucas-manuel) [inline src/proposals/20230830/SparkEthereum_20230830.sol:6]: Dependency imports above local imports

File: src/proposals/20230830/SparkEthereum_20230830.sol [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    function capsUpdates()
+        public view override returns (IEngine.CapsUpdate[] memory)
+    {
+        IEngine.CapsUpdate[] memory capsUpdate = new IEngine.CapsUpdate[](1);
+        capsUpdate[0] = IEngine.CapsUpdate({
+            asset: wstETH,
+            supplyCap: NEW_WSTETH_SUPPLY_CAP,
+            borrowCap: EngineFlags.KEEP_CURRENT
```
Comment (lucas-manuel) [inline src/proposals/20230830/SparkEthereum_20230830.sol:53]: Align

File: src/proposals/20230830/SparkEthereum_20230830.t.sol [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+    uint256 public   constant OLD_WETH_OPTIMAL_USAGE_RATIO   = 0.80e27;
+    uint256 public   constant NEW_WETH_OPTIMAL_USAGE_RATIO   = 0.90e27;
+    uint256 public   constant OLD_WETH_VARIABLE_RATE_SLOPE_1 = 0.030e27;
+    uint256 public   constant NEW_WETH_VARIABLE_RATE_SLOPE_1 = 0.028e27;
+    uint256 public   constant OLD_WETH_VARIABLE_RATE_SLOPE_2 = 0.80e27;
+    uint256 public   constant NEW_WETH_VARIABLE_RATE_SLOPE_2 = 1.20e27;
+
+    address public   constant wstETH                         = 0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0;
+    uint256 public   constant OLD_WSTETH_SUPPLY_CAP          = 200_000;
+    uint256 public   constant NEW_WSTETH_SUPPLY_CAP          = 400_000;
```
Comment (lucas-manuel) [inline src/proposals/20230830/SparkEthereum_20230830.t.sol:22]: Rm extra spaces between public and constant

Comment (lucas-manuel) [reply src/proposals/20230830/SparkEthereum_20230830.t.sol:22]: Group addresses and uints

File: src/proposals/20230830/SparkEthereum_20230830.t.sol [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+
+    constructor() {
+        id = '20230830';
+    }
+
+    function setUp() public {
+        vm.createSelectFork(getChain('mainnet').rpcUrl, 17_935_850);
```
Comment (lucas-manuel) [inline src/proposals/20230830/SparkEthereum_20230830.t.sol:29]: Lets comment this out until the spell is deployed

File: src/proposals/20230830/SparkEthereum_20230830.t.sol [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+        /*******************************************/
+        
+        ReserveConfig memory wstETHConfigBefore = _findReserveConfig(allConfigsBefore, wstETH);
+        assertEq(wstETHConfigBefore.supplyCap, OLD_WSTETH_SUPPLY_CAP);
+
+        /*****************************************************/
+        /*** WETH Interest Rate Strategy Before Assertions ***/
+        /*****************************************************/
+        
+        ReserveConfig memory WETHConfigBefore = _findReserveConfig(allConfigsBefore, WETH);
```
Comment (lucas-manuel) [inline src/proposals/20230830/SparkEthereum_20230830.t.sol:51]: wethConfigBefore

File: src/proposals/20230830/SparkEthereum_20230830.t.sol [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+        IDefaultInterestRateStrategy interestRateStrategyBefore = IDefaultInterestRateStrategy(
```
Comment (lucas-manuel) [inline src/proposals/20230830/SparkEthereum_20230830.t.sol:52]: Interest rate strategy isnt changing so don't need `Before`

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
First round of comments, looking good so far.

Also lets move the folder from the last proposal into `archive` director now.

File: src/proposals/20230830/SparkEthereum_20230830.sol [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+import { SparkPayloadEthereum, IEngine, Rates, EngineFlags } from '../../SparkPayloadEthereum.sol';
+
+import { IPoolAddressesProvider } from 'aave-v3-core/contracts/interfaces/IPoolAddressesProvider.sol';
+
+/**
+ * @title  August 30, 2023 Spark Ethereum Proposal - Update ETH Kink, Update ETHn Kink Rate, Update wstETH Supply Cap
+ * @author Phoenix Labs
+ * @dev    This proposal updates ETH market parameters and raises wstETH supply cap
+ * Forum:  https://forum.makerdao.com/t/phoenix-labs-proposed-changes-for-spark-for-next-upcoming-spell/21685
+ * Vote:   N/A
```
[the remark this answers is not in the document]
Comment (hexonaut) [reply src/proposals/20230830/SparkEthereum_20230830.sol:13]: Polls: https://vote.makerdao.com/polling/QmULJKwK and https://vote.makerdao.com/polling/QmbMR8PU

[the remark this answers is not in the document]
Comment (barrutko) [reply src/proposals/20230830/SparkEthereum_20230830.sol]: Added ✅

File: src/proposals/20230830/SparkEthereum_20230830.t.sol (hunk 9 above)
Comment (barrutko) [reply src/proposals/20230830/SparkEthereum_20230830.t.sol:29]: Not sure what you mean. Can you elaborate?

Comment (hexonaut) [reply src/proposals/20230830/SparkEthereum_20230830.t.sol:29]: He means it's good to remove pinning to a specific block until the spell is deployed.

File: src/proposals/20230830/SparkGoerli_20230830.t.sol [hunk 13]
```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.10;
+
+import '../../SparkTestBase.sol';
+
+import { Ownable } from 'aave-v3-core/contracts/dependencies/openzeppelin/contracts/Ownable.sol';
```
Comment (hexonaut) [inline src/proposals/20230830/SparkGoerli_20230830.t.sol:6]: Unused.

**Review by hexonaut — CHANGES_REQUESTED** [review]
Small adjustments, but overall looks good. Checked:

 * Hardcoded addresses are correct
 * Values are correct
 * Code is correct
 * Goerli spell matches mainnet
 * Tests are correct
 * Tests pass
 * Goerli spell tests matches mainnet
 * Diff values are all correct

File: src/proposals/20230830/SparkEthereum_20230830.t.sol (hunk 9 above)
Comment (barrutko) [reply src/proposals/20230830/SparkEthereum_20230830.t.sol:27]: Ok, got it

**Review by hexonaut — APPROVED** [review]
Good to deploy.

File: src/proposals/20230830/SparkGoerli_20230830.t.sol [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+                baseStableBorrowRate:          NEW_WETH_VARIABLE_RATE_SLOPE_1,
+                stableRateSlope1:              interestRateStrategy.getStableRateSlope1(),
+                stableRateSlope2:              interestRateStrategy.getStableRateSlope2(),
+                baseVariableBorrowRate:        interestRateStrategy.getBaseVariableBorrowRate(),
+                variableRateSlope1:            NEW_WETH_VARIABLE_RATE_SLOPE_1,
+                variableRateSlope2:            NEW_WETH_VARIABLE_RATE_SLOPE_2
+            })
+        );
+    }
+}
```
Comment (lucas-manuel) [inline src/proposals/20230830/SparkGoerli_20230830.t.sol:104]: Add new line at the end of file: https://stackoverflow.com/questions/44704968/visual-studio-code-insert-newline-at-the-end-of-files

File: src/proposals/20230830/SparkEthereum_20230830.t.sol [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+        );
+        
+        /***********************/
+        /*** Execute Payload ***/
+        /***********************/
+        
+        GovHelpers.executePayload(vm, payload, executor);
+
+        ReserveConfig[] memory allConfigsAfter = createConfigurationSnapshot('', pool);
+
```
Comment (lucas-manuel) [inline src/proposals/20230830/SparkEthereum_20230830.t.sol:74]: Should configure this in your VS Code as well: https://linux.how2shout.com/remove-trailing-spaces-automatically-in-visual-code-studio/ the only diff between the files is whitespace

**Review by lucas-manuel — DISMISSED** [review]
Nice PR, LGTM. 

Left some comments for VS Code configurations to make PRs consistent, highly recommend getting those set up.

- Reviewed mainnet addresses against aug 16 spell
- Reviewed goerli addresses against output/5/primary-latest.json
- Reviewed diff file against forum post
- Diffed spells against each other
- Diffed tests against each other
- Checked logic of spell, compared values against forum post
- Checked logic of test, compared values against forum post and diff file

**Review by hexonaut — DISMISSED** [review]
Good to deploy.

**Review by lucas-manuel — DISMISSED** [review]

**Review by hexonaut — APPROVED** [review]
LGTM once the code verifies, however we don't want to merge this until it goes into the main spell.

**Review by lucas-manuel — APPROVED** [review]

**Comment by lucas-manuel** [comment]
LGTM, lets hold off on merge for now.

**Comment by hexonaut** [comment]
This can be merged now that Maker spell is live on portal.

**Comment by lucas-manuel** [comment]
Conflicts need to be resolved before merge @barrutko

File: src/AaveV3PayloadBase.sol [hunk 16]
```diff
@@ -3,6 +3,7 @@ pragma solidity ^0.8.0;
 
 import {Address} from './libraries/Address.sol';
 import {IAaveV3ConfigEngine as IEngine} from './interfaces/IAaveV3ConfigEngine.sol';
+import {IV3RateStrategyFactory as Rates} from './interfaces/IV3RateStrategyFactory.sol';
```
Comment (barrutko) [inline src/AaveV3PayloadBase.sol:6]: Had to add this

**Review by hexonaut — APPROVED** [review]
Good to merge.

**Review by lucas-manuel — APPROVED** [review]

**Pull request merged**